operator/pkg/operator.go

```diff
@@ -194,6 +274,7 @@ func (o *Operator) Start(ctx context.Context) error {
 func (o *Operator) handleNewBatchLogV2(newBatchLog *servicemanager.ContractAlignedLayerServiceManagerNewBatchV2) {
 	o.Logger.Infof("Received new batch log V2")
 	err := o.ProcessNewBatchLogV2(newBatchLog)
+	o.UpdateLastProcessBatch(newBatchLog.TaskCreatedBlock)
```

Comment:
This should happen only after we finish verifying. Possibly `defer` this.

Reply:
Also, the way this is updated looks like a data race. We should probably write a buffered channel instead, so a single goroutine updates the value and writes the file. Possibly as part of the `select` on the main loop.

Reply:
Yeah that's a data race 100%